crates/miden-objects/src/block/proven_block.rs

```diff
@@ -150,21 +134,15 @@ impl ProvenBlock {
 impl Serializable for ProvenBlock {
     fn write_into<W: ByteWriter>(&self, target: &mut W) {
         self.header.write_into(target);
-        self.updated_accounts.write_into(target);
-        self.output_note_batches.write_into(target);
-        self.created_nullifiers.write_into(target);
-        self.transactions.write_into(target);
+        self.body.write_into(target);
     }
 }
 
 impl Deserializable for ProvenBlock {
     fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
         let block = Self {
             header: BlockHeader::read_from(source)?,
-            updated_accounts: <Vec<BlockAccountUpdate>>::read_from(source)?,
-            output_note_batches: <Vec<OutputNoteBatch>>::read_from(source)?,
-            created_nullifiers: <Vec<Nullifier>>::read_from(source)?,
-            transactions: OrderedTransactionHeaders::read_from(source)?,
+            body: BlockBody::read_from(source)?,
         };
```

Comment:
Not from this PR, but should we leave a comment here and in `SignedBlock` deserialization code explaining why we do not validate the consistency of header and body? I assume it's because validation is going to be reasonably expensive (ZKP verification, signature verification), and so they will be done explicitly?
Still feels a bit dangerous to potentially deserialize an inconsistent block.

Reply:
Validating a signature should be relatively cheap, but I'm assuming this will happen within `BlockHeader` itself (since the signature will be in the block header).

Validating the rest would be relatively expensive:

- Verifying the ZKP itself will probably be 10 - 20 ms.
- We'll also need to compute the output note tree root, transaction commitment, account update commitment, and nullifier commitment. Depending on the number of transactions and their complexity, this could also be pretty significant (probably also on the order of 10 - 20 ms - though, would be good to test this at some point).

Not in this PR, but we should probably add `ProvenBlock::validate()` method to make sure the structure is consistent. Let's create an issue for this (unless we have one already).